backend/api/submissions/permissions.py

```diff
@@ -27,12 +28,10 @@ def has_permission(self, source, info):
         if user_info.is_staff or source.speaker_id == user_info.id:
             return True
```

Comment:
can you move all those if(s) check that don't do queries above the schedule_items?

I am talking about:

```
if not pastaporto.is_authenticated:
```

and

```
        if user_info.is_staff or source.speaker_id == user_info.id:
```

seems wasteful to run the schedule_items query if we are then going to pass the Permission because they are staff or reject because they are unauthenticated

Reply:
I don't think that is right, because `if source.schedule_items.exists()` means the submission is public, and so anyone can see it because it's in the Schedule, even if they don't have an account and are not authenticated. 
If I swap those checks, I will force them to have an account... doesn't seem right to me...

Reply:
Any thoughts on this?

Reply:
>  If I swap those checks, I will force them to have an account... doesn't seem right to me...

Ideally what I wanted was that the code was smart enough that if you are logged in > do the easy check that don't require the DB first

If you are not logged in and it is not public, fail 😃 


But don't worry, it is a minor optimization and we can do it in another PR